src/sisl/io/vasp/chg.py

```diff
@@ -72,11 +72,15 @@ def read_grid(self, index=0, dtype=np.float64, **kwargs):
             i = len(vals)
 
             if i % n == 0 and i < n * max_index:
-                # Each time a new spin-index is present, we need to read the coordinates
-                j = 0
-                while j < geom.na:
-                    j += len(rl().split())
-
+                if self.base_file.startswith("CHGCAR"):
```

Comment:
how does this work with *other* files which are not a `CHGCAR`. And is this check even needed? I mean users could do: `mv CHGCAR run1.CHGCAR` and sisl will still get the correct type, but this check will fail. Is it necessary with this ending check?

Reply:
What is really needed is a check that distinguishes CHG and CHGCAR. As I zipped the files I decided to use `startswith`. But you're right, if the user changes the filename this is not a good way to do the check.

Reply:
I've found another way to distinguish the two file types. CHG is 10 columns while CHGCAR has 5.